docker.Makefile

```diff
@@ -8,32 +8,54 @@ $(warning WARNING: No docker user found using results from whoami)
 DOCKER_ORG                = $(shell whoami)
 endif
 
-CUDA_VERSION              = 11.3
-CUDNN_VERSION             = 8
+CUDA_VERSION_FULL         = 11.3.1
+CUDA_VERSION_MAJOR_MINOR  = $(shell echo $(CUDA_VERSION_FULL) | cut -d. -f1-2)
 BASE_RUNTIME              = ubuntu:18.04
-BASE_DEVEL                = nvidia/cuda:$(CUDA_VERSION)-cudnn$(CUDNN_VERSION)-devel-ubuntu18.04
+ifeq ("$(CUDA_VERSION_FULL)","cpu")
+MUTEX_PACKAGE             = cpuonly
+BASE_DEVEL                = $(BASE_RUNTIME)
+else
+CUDNN_VERSION             = 8
+MUTEX_PACKAGE             = cudatoolkit=$(CUDA_VERSION_MAJOR_MINOR)
+BASE_DEVEL                = nvidia/cuda:$(CUDA_VERSION_FULL)-cudnn$(CUDNN_VERSION)-devel-ubuntu18.04
+endif
 
 # The conda channel to use to install cudatoolkit
 CUDA_CHANNEL              = nvidia
 # The conda channel to use to install pytorch / torchvision
-INSTALL_CHANNEL           = pytorch
+INSTALL_CHANNEL          ?= pytorch
 
-PYTHON_VERSION            = 3.8
-PYTORCH_VERSION           = $(shell git describe --tags --always)
+PYTHON_VERSION           ?= 3.8
+PYTORCH_VERSION          ?= $(shell git describe --tags --always)
```

Comment:
Why make these sets optional? Do we expect all these variables to have a default values set?

Reply:
Making these optional allows us to have sane defaults and also make it so that we can override them with environment variables within CI

Reply:
Otherwise we'd have to pass them as make variables :(